superset/db_engine_specs/presto.py

```diff
@@ -911,10 +913,18 @@ def where_latest_partition(  # pylint: disable=too-many-arguments
         if values is None:
             return None
 
-        column_names = {column.get("name") for column in columns or []}
+        column_type_by_name = {
+            column.get("name"): column.get("type") for column in columns or []
+        }
+
         for col_name, value in zip(col_names, values):
-            if col_name in column_names:
-                query = query.where(Column(col_name) == value)
+            if col_name in column_type_by_name:
+                if column_type_by_name.get(col_name) == "TIMESTAMP":
+                    query = query.where(Column(col_name, TimeStamp()) == value)
+                elif column_type_by_name.get(col_name) == "DATE":
+                    query = query.where(Column(col_name, Date()) == value)
+                else:
+                    query = query.where(Column(col_name) == value)
```

Comment:
I'm wondering if we couldn't just use the existing `convert_dttm` here, as we should already have type casting logic in place for these types of conversions? https://github.com/apache/superset/blob/8ab45c90ccc04f51a0e7db12b622fb63a29ec27c/superset/db_engine_specs/presto.py#L597-L604

Something like `query = query.where(Column(col_name) == cls.convert_dttm(column_type_by_name.get(col_name), value)` (not tested! thinking out loud)

Reply:
Certainly possible but the drawbacks I see to re-using this existing method is that: 

1. We have to explicitly cast the string `value` to a python `datetime` to call the method only to get it converted back to a string with `.isoformat`.
2. `from_iso8601_x` doesn't seem to be a recommended function anymore in the latest versions of Presto/Trino https://trino.io/docs/current/language/types.html#date-and-time
3. This method only works if your date/timestamps are in ISO 8601 format. 
4. In the future, if more Trino types are to be supported in previewing data with partitions, we will have to create new types like this anyways. Also, I think this is more the sqlalchemy friendly way.